src/Aspire.Hosting.Docker/DockerComposeEnvironmentResource.cs

```diff
@@ -347,9 +358,9 @@ private async Task PrepareAsync(PipelineStepContext context)
                 defaultValue = await parameter.GetValueAsync(context.CancellationToken).ConfigureAwait(false);
             }
 
-            if (envVar.Source is ContainerImageReference cir && cir.Resource.TryGetContainerImageName(out var imageName))
+            if (envVar.Source is ContainerImageReference && envVar.Source is IValueProvider cirValue)
             {
-                defaultValue = imageName;
+                defaultValue = await cirValue.GetValueAsync(context.CancellationToken).ConfigureAwait(false);
```

Comment:
```suggestion
            if (envVar.Source is ContainerImageReference cir)
            {
                defaultValue = await ((IContainerRegistry)cir).GetValueAsync(context.CancellationToken).ConfigureAwait(false);
```

This reads a little cleaner IMO. Double type checking is kind of weird.